### ISSUE TITLE
Close the FEEL popup editor internally

### ISSUE THREAD

**Issue by marcellobarile** [camunda/camunda-bpmn-js#312]
__Is your feature request related to a problem? Please describe.__

Right now, the logic to close the FEEL popup editor is, for certain cases, demand to the consumer which can lead to unexpected side effects (e.g. https://github.com/camunda/web-modeler/issues/6334).

Moreover, for some reason, the `.close()` method does not work all the time (maybe it doesn't work when some parent is removed from the DOM?), also in this case causing weird side effects (the `close()` gets called but nothing happens).

__Describe the solution you'd like__

I would expect that the FEEL popup editor gets automatically closed when not necessary anymore:

* when the modeler gets disposed or unmounted
* when the properties panel (technical properties editing) gets unmounted
* when the property being edited is no longer available

__Describe alternatives you've considered__

An alternative could be closing it forcefully so that the `close()` method would always have the same effect.

__Additional context__

This topic was raised up from a discussion in this thread: https://camunda.slack.com/archives/CKGH9LR40/p1696495030377929

Other approaches were used, for example providing the `propertiesPanel.feelPopupContainer`, but this one was not working as well, probably because the DOM node was gone before the actual close could take place.

What worked was calling the `close` method right before doing the `this.modeler.destroy()` in the web-modeler component in charge of rendering the bpmn view.

```javascript
    if (this.modelerInstance) {
      this.unbindEvents();

      try {
        const feelPopup = this.modeler.get('feelPopup');
        feelPopup.close();
      } catch (err) {
        console.log(err);
      }

      this.modeler.destroy();
    }
```

**Comment by marcellobarile** [comment]
Side note:
we might have other corner cases where the FEEL editor remains open and the above workaround doesn't work;

for example, when switching mode in the web-modeler (e.g. from `implement` to `design`).
In this case, the modeler doesn't get disposed, because the `bpmn viewer` component is still there, but the `properties panel` changes radically (we only have comments in the `design` view); **the FEEL editor remains open even if there is no field anymore in the properties panel.**

**Comment by nikku** [comment]
@marcellobarile What is your expectation when switching between modes? I'd assume that the panel closes as the properties panel (technical editing) is gone.

**Comment by marcellobarile** [comment]
@nikku correct :)

**Comment by nikku** [comment]
@marcellobarile Thanks. I updated the issue accordingly.

**Comment by nikku** [comment]
I've tagged this as `spring cleaning` for the moment. Let's re-visit this once we reworked the popup menu, cf. https://github.com/camunda/camunda-modeler/issues/3902.

**Comment by nikku** [comment]
Will be fixed via `@bpmn-io/properties-panel@3.17.0`.

**Comment by marcellobarile** [comment]
@nikku we have this workaround in place:
https://github.com/camunda/web-modeler/blob/9b1bb5839489e1bb8295844472bbec6f11bee71c/webapp/src/components/Modeler/bpmn/BpmnRenderer.js#L224C7-L230C8

Do you think it will still be working after the change introduced with `3.17.0` or do we risk any regression?
We have the `try..catch` that should act as a safeguard, but better safe than sorry :D
